### DISCUSSION TITLE
Changing innerHTML of rendered Slate div

### DISCUSSION THREAD

**Discussion by zharramadar** [ianstormtaylor/slate#4830] [Q&A]
I am writing a chrome extension that needs to change the text in a contenteditable div that has been rendered using Slate. If I just set it as `element.innerHTML = '<div>blablabla</div>'`, I can see the change taking place, but as soon as the box loses its focus, it'll reload the previous content (in that particular page case, I believe it is fetching data from the server again). 
Is there a way to change that innerHTML while making the component understand this is a valid change and it should keep the text and save instead of replacing it.
Or even, is there a way to get a slate editor object from that rendered div? I thought I might be able to add Slate into my extension and manipulate the rich text box using Slate's API if I could grab the object for that particular contenteditable div.

**Comment by TheSpyder** [comment]
> I can see the change taking place, but as soon as the box loses its focus, it'll reload the previous content (in that particular page case, I believe it is fetching data from the server again).

More likely React is simply re-rendering Slate's internal data. Mutating the DOM does nothing permanent in a VDOM environment. What you'll need to do is dig into the instance of React controlling the page and find the Slate component, then set content through it. I have no idea how to do this but I think it should be possible.

**Reply by zharramadar** [reply]
my main concern is, as I'm in an extension, I might not have access to any JS the original page is outputting, as extension code is sandboxed. I don't think it is possible to get the object the page created the control with, thus my question about creating an Slate object from an existing element. 
I thought it would be fetching from the server as I see a network request when the focus is lost, and I removed any instances in the page where the text could be found in hidden dom. Maybe Slate is just caching this somewhere else?

Anyway, if anyone knows how can I change the slate object's text outside of slate, or even if it is possible, it would be appreciated.

**Reply by TheSpyder** [reply]
You can already navigate the DOM, you'll be able to navigate the window JS objects. While there may be some fetch from the server, Slate isn't caching the content - React is.

React isn't magic, I've seen (and used) extensions that can patch pages written in React by looking up the component hierarchy and changing things. I just don't know how they work, I don't use React in my day job.

**Reply by PhilBookst** [reply]
Hey @zharramadar did you get this working?
I'm currently trying to achieve the same thing - I need to modify the textContent of a slate node from the DOM, without having access to the slate react component
